src/app/payment/mapSubmitOrderErrorMessage.ts

```diff
@@ -27,6 +28,14 @@ export default function mapSubmitOrderErrorMessage(
                 return translate('payment.payment_method_error', { message: error.message });
             }
 
+            if (shouldLocalise && error.body && error.body.errors) {
+                const messages = error.body.errors.map((err: { code: any }) => translate(`payment.errors.${err.code}`));
+
+                if (messages.length) {
```

Comment:
is it possible that you might be trying to translate a key that doesn't exist? I'm not sure if we currently have a way to detect if a key is present.

Probably a hack we could do is
```
return message === `payments.error.${err.code}` ? err.message : message;
```

This way if there was no translation found, we default to the English text. This might be especially useful for checkouts that haven't provided a translation.

Wonder if it's worth having it built-in in the `translate` function

```
// translate(id, data, default)
// translate(`payment.errors.${err.code}`, {}, err.message)
```

Thoughts?

Reply:
If there's no messages, we don't want it to return an empty string. We want to go to the next branch to display `error.message` if exists. We don't want to end up with an empty modal.

Reply:
@capsula4 ... We currently don't have a way to detect if the key exists in the json file or not. What locale shall we check?

Currently, when a key does not exist, the missing key is logged as warning so we detect it and add it, and the key is displayed instead of the message. Changes to this behaviour has to be in `translate()` function in `LanguageService` in checkout-sdk